Sources/NIOHTTP1Client/main.swift

```diff
@@ -0,0 +1,100 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHTTP1
+import NIOOpenSSL
+import Foundation
+
+private final class HTTPResponseHandler: ChannelInboundHandler {
+
+    let promise: EventLoopPromise<Void>
+
+    init(_ promise: EventLoopPromise<Void>) {
+        self.promise = promise
+    }
+
+    typealias InboundIn = HTTPClientResponsePart
+
+    func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+        let httpResponsePart = unwrapInboundIn(data)
+        switch httpResponsePart {
+        case .head(let httpResponseHeader):
+            print("\(httpResponseHeader.version) \(httpResponseHeader.status.code) \(httpResponseHeader.status.reasonPhrase)")
+            for (name, value) in httpResponseHeader.headers {
+                print("\(name): \(value)")
+            }
+        case .body(var byteBuffer):
+            if let responseBody = byteBuffer.readString(length: byteBuffer.readableBytes) {
+                print(responseBody)
+            }
+        case .end(_):
+            _ = ctx.channel.close()
+            promise.succeed(result: ())
+        }
+    }
+
+    func errorCaught(ctx: ChannelHandlerContext, error: Error) {
+        print("Error: ", error)
+        _ = ctx.channel.close()
```

Comment:
same here, `ctx.channel.close(promise: nil)`

Reply:
I changed in 10c09d5.